source/Nuke.SourceGenerators.Tests/AccessInheritanceAsMembersGeneratorTest.Test#AccessInheritanceAsMembersGenerator.verified.cs

```diff
@@ -0,0 +1,9 @@
+﻿//HintName: AccessInheritanceAsMembersGenerator.cs
+// <auto-generated/>
+using System;
+
+internal partial class Build
+{
+    protected IDisposable IDisposable  => this;
```

Comment:
Not clear to me why `IDisposable` should show up here. I'd filter to interfaces that implement `INukeBuild`.

Reply:
Maybe I didn't understood the issue correctly.

I thought each class (like Build) that inherits from INukeBuild should have all interfaces (no matter what they are inheriting) as members.

But maybe what you are expecting is that the IHazRepository of the example inherits INukeBuild and that is the reason, why IHazRepository should be a member of the Build class.

If this is the case, this wasn't clear to me, as I thought the INukeBuild is a marker for the Build class.

Maybe, you can extend the example and the description a little bit :-)